modules/ml/src/main/java/org/apache/ignite/ml/naivebayes/gaussian/GaussianNaiveBayesModel.java

```diff
@@ -0,0 +1,110 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.naivebayes.gaussian;
+
+import java.io.Serializable;
+import org.apache.ignite.ml.Exportable;
+import org.apache.ignite.ml.Exporter;
+import org.apache.ignite.ml.Model;
+import org.apache.ignite.ml.math.primitives.vector.Vector;
+
+/**
+ * Simple naive Bayes model which predicts result value {@code y} belongs to a class {@code C_k, k in [0..K]} as {@code
+ * p(C_k,y) = p(C_k)*p(y_1,C_k) *...*p(y_n,C_k) / p(y)}. Return the number of the most possible class.
+ */
+public class GaussianNaiveBayesModel implements Model<Vector, Double>, Exportable<GaussianNaiveBayesModel>, Serializable {
+    /** */
+    private static final long serialVersionUID = -127386523291350345L;
+    /** Means of features for all classes. kth row contains means for labels[k] class. */
+    private final double[][] means;
+    /** Variances of features for all classes. kth row contains variances for labels[k] class */
+    private final double[][] variances;
+    /** Prior probabilities of each class */
+    private final double[] classProbabilities;
+    /** Labels. */
+    private final double[] labels;
+    /** feature sum, squared sum and cound per label */
+    private GaussianNaiveBayesSumsHolder sumsHolder;
+
+    /**
+     * @param means means of features for all classes
+     * @param variances variances of features for all classes
+     * @param classProbabilities probabilities for all classes
+     * @param labels labels
+     * @param sumsHolder feature sum, squared sum and count sum per label. This data is used for future model updating.
+     */
+    public GaussianNaiveBayesModel(double[][] means, double[][] variances,
+        double[] classProbabilities, double[] labels, GaussianNaiveBayesSumsHolder sumsHolder) {
+        this.means = means;
+        this.variances = variances;
+        this.classProbabilities = classProbabilities;
+        this.labels = labels;
+        this.sumsHolder = sumsHolder;
+    }
+
+    /** {@inheritDoc} */
+    @Override public <P> void saveModel(Exporter<GaussianNaiveBayesModel, P> exporter, P path) {
+        exporter.save(this, path);
+    }
+
+    /** Returns a number of class to which the input belongs. */
+    @Override public Double apply(Vector vector) {
+        int k = classProbabilities.length;
+
+        double maxProbapility = .0;
+        int max = 0;
+
+        for (int i = 0; i < k; i++) {
+            double p = classProbabilities[i];
+            for (int j = 0; j < vector.size(); j++) {
```

Comment:
Could you explain what do you mean?

Reply:
In case of sparce vector it would be better to iterate only over the not-null values. Perhaps @avplatonov can say a bit more. At the same time I'm not sure if we have any algorithm that supports sparse vectors good enough so never mind.